File: rest-api/src/com/espe/cicte/ws/service/IlusionService.java
package com.espe.cicte.ws.service;

import javax.ws.rs.Consumes;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import com.espe.cicte.ws.db.IlusionDB;
import com.espe.cicte.ws.dto.IlusionRequest;
import com.espe.cicte.ws.dto.IlusionResponse;


@Path("/cicte/ilusion")
public class IlusionService {

	private static IlusionDB IlusionDB = new IlusionDB();

	@POST
	@Path("/addIlusion")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public IlusionResponse addIlusion(IlusionRequest request) {
		IlusionResponse response = new IlusionResponse();

		response = IlusionDB.addIlusion(request);
		
		return response;
	}
	
	@POST
	@Path("/getIlusion")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public IlusionResponse getIlusion(IlusionRequest request) {
		IlusionResponse response = new IlusionResponse();

		response = IlusionDB.getIlusion();
	
		return response;
	}
	
	@POST
	@Path("/getIlusionByName")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public IlusionResponse getIlusionName(IlusionRequest request) {
		IlusionResponse response = new IlusionResponse();

		response = IlusionDB.getIlusionByName(request);
	
		return response;
	}
	
	@POST
	@Path("/getIlusionById")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public IlusionResponse getIlusionById(IlusionRequest request) {
		IlusionResponse response = new IlusionResponse();

		response = IlusionDB.getIlusionById(request);
	
		return response;
	}
	
	@POST
	@Path("/updateIlusion")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public IlusionResponse updateIlusion(IlusionRequest request) {
		IlusionResponse response = new IlusionResponse();

		response = IlusionDB.updateIlusion(request);

		return response;
	}
	
	@POST
	@Path("/deleteIlusion")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public IlusionResponse deleteIlusion(IlusionRequest request) {
		IlusionResponse response = new IlusionResponse();

		response = IlusionDB.deleteIlusion(request);

		return response;
	}
}

File: cicte-app/src/app/services/ilusion.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable, throwError } from 'rxjs';
import { map, retry } from 'rxjs/operators';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})

export class IlusionService {
  API_URL = environment.API_URL;
  constructor(public http: HttpClient) {
  }
  getQuery( query: string, request: any) {
    const url = `${this.API_URL}${ query }`;
    const headers = new HttpHeaders({
      'Content-Type': 'application/json'
    });
    const body = JSON.stringify(request);
    console.log('Paramentros enviados: ' + body );
    return this.http.post(url, body, { headers: headers });
  }

  getIlusion(ilusion: any) {
    return this.getQuery('ilusion/getIlusion', ilusion)
                  .pipe( map(data => data['ilusion']));
  }
  getIlusionById(ilusion: any) {
    return this.getQuery('ilusion/getIlusionById', ilusion)
                  .pipe( map(data => data['ilusion']));
  }
  getIlusionByName(ilusion: any) {
    return this.getQuery('ilusion/getIlusionByName', ilusion)
                  .pipe( map(data => data['ilusion']));
  }
  addIlusion(ilusion: any) {
    return this.getQuery( 'ilusion/addIlusion', ilusion )
                  .pipe( map(data => data));
  }
  updateIlusion(ilusion: any) {
    return this.getQuery( 'ilusion/updateIlusion', ilusion )
                  .pipe( map(data => data));
  }
  deleteIlusion(ilusion: any) {
    return this.getQuery('ilusion/deleteIlusion', ilusion)
                  .pipe( map(data => data));
  }
}
